src/main/java/org/reactome/server/service/controller/citation/CitationController.java

```diff
@@ -0,0 +1,294 @@
+            out.close();
+            in.close();
+        }
+    }
+
+
+
+    private Citation getCitationFromMap(Map<String, Object> map) {
+        Citation citation = new Citation();
+        citation.setId(map.containsKey("id") ? map.get("id").toString() : map.get("stid").toString());
+        citation.setTitle(map.containsKey("title") ? (String)map.get("title") : (String)map.get("pathwayTitle"));
+
+        Map<String, Object> journalInfo = map.containsKey("journalInfo") ? (Map<String, Object>)map.get("journalInfo") : null;
+        Map<String, Object> journal = journalInfo != null && journalInfo.containsKey("journal") ? (Map<String, Object>)journalInfo.get("journal") : null;
+        citation.setJournal(journal != null && journal.containsKey("title") ? journal.get("title").toString() : null);
+        citation.setYear(journalInfo != null && journalInfo.containsKey("yearOfPublication") ? journalInfo.get("yearOfPublication").toString() : map.get("publicationYear").toString());
+        citation.setMonth(journalInfo != null && journalInfo.containsKey("monthOfPublication") ? journalInfo.get("monthOfPublication").toString() : map.get("publicationMonth").toString());
+        citation.setNumber(journalInfo != null && journalInfo.containsKey("issue") ? journalInfo.get("issue").toString() : null);
+        citation.setVolume(journalInfo != null && journalInfo.containsKey("volume") ? journalInfo.get("volume").toString() : null);
+        citation.setIssn(journal != null && journal.containsKey("issn") ? journal.get("issn").toString() : null);
+        citation.setPages(map.containsKey("pages") ? map.get("pageInfo").toString() : null);
+
+        List<String> urls = new ArrayList<>();
+        if(map.containsKey("doi")){
+            String doi = (String)map.get("doi");
+            citation.setDoi(doi);
+            urls.add(DOI_BASE_URL + doi);
+        }
+        else if(map.containsKey("stid")) {
+            urls.add(map.get("baseURL") + "/content/detail/" + map.get("stid"));
+        }
+        citation.setUrls(urls);
+
+        if(map.containsKey("authorList")) {
+            Map<String, Object> authorList = (Map<String, Object>) map.get("authorList");
+            citation.setAuthors(authorList.containsKey("author") ? (List<Map<String, String>>)authorList.get("author") : null);
+        }
+        else if(map.containsKey("authors")) {
+            citation.setAuthors((List<Map<String, String>>)map.get("authors"));
+        }
+
+        citation.setPathway((Boolean)map.get("isPathway"));
+        return citation;
+    }
+
+    // private method for getting the citation details for a static citation
+    // for export purposes
+    private Map<String, Object> getStaticCitationForExport(String id) {
+        try {
+            HttpGet request = new HttpGet(new URIBuilder()
+                    .setScheme("https")
+                    .setHost("www.ebi.ac.uk")
+                    .setPath("/europepmc/webservices/rest/search")
```

Comment:
Since this endpoint is used twice in this class. You can create a static final together with the attributes.
`private static String EUROPE_PMC_ENDPOINT = "https://www.ebi.ac.uk/europepmc/webservices/rest/search";`
then in the `URIBuilder` you can use like `new URIBuilder(EUROPE_PMC_ENDPOINT)`